### PR TITLE
Use submodule to download java-models-library

### PR DESCRIPTION
**Pull request by peterschrammel**

### CODE REVIEW

**Comment by cesaro** [comment]
This won't fix the current CI failures in `develop`. I have fixed that in the last commit of #2280, which I hope to merge today (after I manage to fix the failing tests in TG).

**Comment by cesaro** [comment]
But it's a step in the right direction, indeed

**Comment by tautschnig** [comment]
I have no prior experience on how GitHub displays submodules or submodule updates as PRs: is the set-up such that this PR would set the submodule at some commit, and any updates would be done via PR that just changes the commit that's being pointed to?

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: c00bf6fc944716ff63919b27c824d101d0c1443b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77136993
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

File: jbmc/src/java_bytecode/library/CMakeLists.txt [hunk 1]
```diff
@@ -1,32 +1,2 @@
...
-      "${JBMC_SOURCE_DIR}/java_bytecode/library/src"
-)
-
-find_package(Java REQUIRED)
-include(UseJava)
-
-set(CMAKE_JAVA_COMPILE_FLAGS -sourcepath "src" -d "classes" -XDignore.symbol.file)
-
-# create a target for the executable performing the .jar -> .inc conversion
+# Executable
```
Comment (smowton) [inline jbmc/src/java_bytecode/library/CMakeLists.txt]: Probably keep the comment above?

**Comment by smowton** [comment]
@tautschnig Yes

**Review by smowton — APPROVED** [review]

**Comment by smowton** [comment]
Looks like tests that directly reference the core-models jar location need amending. Also perhaps worth looking at whether it's right that they core dump when the path is wrong, rather than reporting an error cleanly.

**Comment by kroening** [comment]
We'll need to produce a bit of scripting to be still able to distribute tar balls. @tautschnig How does this work with Debian?

**Comment by tautschnig** [comment]
Since there isn't a problem distributing it when the source is made available it's all fine as far as Debian is concerned. I believe the only point of concern is any user shipping binary-only products based on CBMC. Those will want a GPL-free tarball.

**Comment by peterschrammel** [comment]
@smowton, @cesaro, TG bump passes.

**Comment by smowton** [comment]
@peterschrammel I already approved, people who own other code needed.

**Review by Degiorgio — APPROVED** [review]

**Comment by owen-mc-diffblue** [comment]
It looks like this needs @forejtv and/or @thk123's approval.

**Review by forejtv — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: b34e951e52b2e920b71d89196e3099f88568e61f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78496248
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Pull request merged**